Wangscape/noise/ModuleManager.cpp

```diff
@@ -20,31 +72,72 @@ ModuleManager::ModuleManager(const Options & options) :
             for (const auto& t2 : clique)
             {
                 TerrainIDPair tp{t1, t2};
-                if (mBordersHorizontal.find(tp) == mBordersHorizontal.end())
-                    mBordersHorizontal.insert({tp, module::makePlaceholder(mRNG())});
-                if (mBordersVertical.find(tp) == mBordersVertical.end())
-                    mBordersVertical.insert({tp, module::makePlaceholder(mRNG())});
+                if (mHorizontalBorders.find(tp) == mHorizontalBorders.end())
+                {
+                    if (options.defaultModuleGroup)
+                        mHorizontalBorders.insert({tp, loadModuleGroup(p.string())});
+                    else
+                        throw std::runtime_error("Missing vertical module group, and no default module group");
+                }
+                if (mVerticalBorders.find(tp) == mVerticalBorders.end())
+                {
+                    if(options.defaultModuleGroup)
+                        mVerticalBorders.insert({tp, loadModuleGroup(p.string())});
+                    else
+                        throw std::runtime_error("Missing vertical module group, and no default module group");
+                }
             }
     }
+    
 }
```

Comment:
Isn't this constructor too big? I seems that you can extract this `for` loops to a function (or a group of classes).